### PR TITLE
codeql fixes

### PR DESCRIPTION
**Pull request by mangod9**
adding a suppression for a read might not be null-terminated

### CODE REVIEW

File: src/coreclr/hosts/corerun/corerun.hpp [hunk 1]
```diff
@@ -471,6 +471,7 @@ namespace pal
         {
             // We have data. At this point we can likely make a strong decision.
             const char tracer_pid_name[] = "TracerPid:";
+            // CodeQL [SM01932] status is of fixed size and [sizeof(status) -1] is specified for read .
```
Comment (janvorli) [inline src/coreclr/hosts/corerun/corerun.hpp:474]: I wonder if we could instead make it happy by adding `status[bytes_read] = 0;`. We could then remove the zero initialization of the status array, which could also make performance a tiny bit better.

Comment (janvorli) [reply src/coreclr/hosts/corerun/corerun.hpp:474]: We actually do exactly the same thing in pal.cpp:
https://github.com/dotnet/runtime/blob/66e45ef152002a2e1cbea939ded6643dfa48307c/src/coreclr/pal/src/init/pal.cpp#L788-L814

Comment (mangod9) [reply src/coreclr/hosts/corerun/corerun.hpp:474]: ok interesting, will make that change then.

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Pull request merged**